ide/models/files.py

```diff
@@ -168,7 +168,7 @@ class Meta(S3File.Meta):
 
 class ResourceIdentifier(IdeModel):
     resource_file = models.ForeignKey(ResourceFile, related_name='identifiers')
-    resource_id = models.CharField(max_length=100)
+    resource_id = models.CharField(max_length=100, validators=regexes.validator('c_identifier', _("Invalid resource ID.")))
```

Comment:
This isn't quite right; resource IDs are prefixed and so don't have the usual start-of-identifier rules.